railgun/deployment/kic-deployment.template.yaml

```diff
@@ -0,0 +1,1095 @@
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kong
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kong-serviceaccount
+  namespace: kong
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kong-ingress-clusterrole
+  creationTimestamp: null
+rules:
+- apiGroups: ["*"]
+  resources: ["*"]
+  verbs: ["*"]
+- nonResourceURLs: ['*']
+  verbs: ['*']
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kong-ingress-clusterrole-nisa-binding
+roleRef:
+  apiGroup: 'rbac.authorization.k8s.io'
+  kind: ClusterRole
+  name: kong-ingress-clusterrole
+subjects:
+- kind: ServiceAccount
+  name: kong-serviceaccount
+  namespace: kong
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: kong-ingress-controller
+  name: kong-ingress-controller
+  namespace: kong
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kong-ingress-controller
+  template:
+    metadata:
+      annotations:
+        kuma.io/gateway: enabled
+        prometheus.io/port: "8100"
+        prometheus.io/scrape: "true"
+        traffic.sidecar.istio.io/includeInboundPorts: ""
+      labels:
+        app: kong-ingress-controller
+    spec:
+      containers:
+      - env:
+        - name: CONTROLLER_KONG_ADMIN_PUBLISH_SERVICE
+          value: "{{namespace/kong-admin-api-service-name}}"
+          # example value: kong-system/ingress-controller-kong-admin
+        - name: CONTROLLER_KONG_ADMIN_TLS_SKIP_VERIFY
+          value: "true"
+        - name: CONTROLLER_PUBLISH_SERVICE
+          value: "{{namespace/kong-proxy-service-name}}"
+          # example value: kong-system/ingress-controller-kong-admin
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        image: "{{kong-ingress-controller-imageurl}}"
+        # example "jimin1/kic-release:verify-fix"
+        imagePullPolicy: IfNotPresent
+        livenessProbe:
+          failureThreshold: 3
+          httpGet:
+            path: /healthz
+            port: 10254
+            scheme: HTTP
+          initialDelaySeconds: 5
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 1
+        name: ingress-controller-kong-controller
+        ports:
+        - containerPort: 8080
+          name: webhook
+          protocol: TCP
+        readinessProbe:
+          failureThreshold: 3
+          httpGet:
+            path: /healthz
+            port: 10254
+            scheme: HTTP
+          initialDelaySeconds: 5
+          periodSeconds: 10
+          successThreshold: 1
+          timeoutSeconds: 1
+      serviceAccountName: kong-serviceaccount
```

Comment:
blocker: please _do not include this new deployment mechanism in this PR_, https://github.com/Kong/kubernetes-ingress-controller/issues/1480 is a `POST KIC 2.0` item that we need to discuss as a team. The team needs to have consensus that we want to support yet another deployment mechanism, and we shouldn't be adding any scope to KIC 2.0 that doesn't have ties to a direct product/customer request or issue.

Reply:
This is not a new mechanism at all. You can take it as parity - user deploy v1.3.x kong (both controller and proxy) using one yaml https://bit.ly/k4k8s.

In 2.0, we can either deploy controller along with proxy or individually. From long term, the benefits of individual yaml is pretty straightforward 
- aligned split kong-proxy/admin into kong-system
- less dependency/loose couply, if proxy fails to be up, controller does not get impacted
- scalable, resources /cpu/memory/db/network more configurable

btw, yaml deployment is generic k8s deployment.

Reply:
Gotcha, thanks for that explanation. That functionality (the `all-in-one` yaml) is actually just using our Kustomize build:

https://github.com/Kong/kubernetes-ingress-controller/blob/main/hack/build-single-manifests.sh

If the intention is to recreate that functionality then I feel like we should update that script with our new kustomize files, rather than creating a new manifest we have to maintain and update manually. How does that sound to you what are your thoughts?